### PR TITLE
Switch from rubygem-ffi to fiddle

### PR DESCRIPTION
**Pull request by skazi0**
Fiddle is an FFI implementation which is included in standard Ruby.
Switching from FFI gem to Fiddle removes one dependency without loosing
any functionality.

### CODE REVIEW

File: yast/lib/suse/connect/yast.rb [hunk 1]
```diff
@@ -1,38 +1,39 @@
...
-  attach_function :offline_system_migrations, [:string, :string, :string], :pointer
-  attach_function :upgrade_product, [:string, :string], :pointer
-  attach_function :synchronize, [:string, :string], :pointer
-  attach_function :system_activations, [:string], :pointer
-  attach_function :search_package, [:string, :string, :string], :pointer
+  extend Fiddle::Importer
+  dlload 'libsuseconnect.so'
+  typealias 'string', 'char*'
+
+  #callback type: void log_line(int, string)
```
Comment (lslezak) [inline yast/lib/suse/connect/yast.rb:17]: If it is not needed then I'd remove it completely, avoid dead code...

**Review by lslezak — APPROVED** [review]
Avoiding extra dependency is good! :+1: 
Thanks!

File: yast/lib/suse/connect/yast.rb (hunk 1 above)
Comment (skazi0) [reply yast/lib/suse/connect/yast.rb:17]: This is not old code, just a plain comment. Fiddle doesn't support callback types so everything is passed as void pointer. Without this, future us would need to dig through the code to find out what is the expected function signature.

**Review by djoreilly — APPROVED** [review]
lgtm

File: yast/lib/suse/connect/yast.rb (hunk 1 above)
Comment (lslezak) [reply yast/lib/suse/connect/yast.rb:17]: Ah, I overlooked that, I thought it is part of the previous ffi code...

**Pull request merged**